test/lit/passes/remove-unused-brs.wast

```diff
@@ -466,6 +466,37 @@
     )
   )
 
+  ;; CHECK:      (func $restructure-br_if-condition-invalidates-6 (type $2) (result i32)
+  ;; CHECK-NEXT:  (local $temp i32)
+  ;; CHECK-NEXT:  (block $block (result i32)
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (br_if $block
+  ;; CHECK-NEXT:     (local.get $temp)
+  ;; CHECK-NEXT:     (local.tee $temp
+  ;; CHECK-NEXT:      (i32.const 1)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (local.get $temp)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $restructure-br_if-condition-invalidates-6 (result i32)
+    (local $temp i32)
+    ;; The br value is syntactically identical to the value at the end of the
+    ;; block, however, the local.tee changes that value so we cannot optimize.
+    (block $block (result i32)
+      (drop
+        (br_if $block
+          (local.get $temp)
+          (local.tee $temp
+            (i32.const 1)
+          )
+        )
```

Comment:
Would it be worth it to continue eliminating the `br_if` in this case, but put its children in a block to preserve them?

Reply:
We can preserve the `local.tee`, but that isn't enough:
```wat
(br_if $block
  (local.get $temp)
  (local.tee $temp
    (i32.const 1)
  )
)
(local.get $temp)

!==

(drop
  (local.tee $temp
    (i32.const 1)
  )
)
(local.get $temp)
```
If the local had 0 before, then we'd be returning different values if we branch or not.

We could save the value and condition to locals:

```wat
(local.set $x
  (local.get $temp)
)
(local.set $y
  (local.tee $temp
    (i32.const 1)
  )
)
(select
  (local.get $x)
  (local.get $temp)
  (local.get $y)
)
```
but that isn't helping on size...